app/lib/update_global_activity_analytics.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+# Updates the global activity analytics by fetching data from the endpoint
+# and saving it in the `GlobalActivityAnalytics` model. It supposes to do that
+# every 24h.
+# The threshold is configurable by `GlobalActivityAnalytics::TTL` constant.
+class UpdateGlobalActivityAnalytics
+  FIELDS = {
+    users: {name: "metrics/visitors", options: {sort: "desc"}},
+    launches: {name: "metrics/mobilelaunches"},
+    gospel_presentations: {name: "metrics/event90"},
+    countries: {name: "cm300000683_58877ac221d4771c0d530484"},
+  }.freeze
+
+  DATE_RANGE_TEMPLATE = "%{start_year}-01-01T00:00:00.000/%{end_year}-01-01T00:00:00.000"
+  BASE_QUERY = {
+    "rsid": "vrs_cru1_godtoolsmobileapp",
+    "dimension": "variables/geocountry",
+    "settings": {
+      "countRepeatInstances": true,
+      "limit": 400,
+      "page": 0,
+      "nonesBehavior": "return-nones",
+    },
+    "statistics": {
+      "functions": [
+        "col-max",
+        "col-min",
+      ],
+    },
+  }
+
+  def initialize
+    @analytics = GlobalActivityAnalytics.instance
+    @analytics_url = ENV.fetch("ADOBE_ANALYTICS_REPORT_URL")
+    @company_id = ENV.fetch("ADOBE_ANALYTICS_COMPANY_ID")
+    @client_id = ENV.fetch("ADOBE_ANALYTICS_CLIENT_ID")
+    @jwt_token = ENV.fetch("ADOBE_ANALYTICS_JWT_TOKEN")
+    @client_secret = ENV.fetch("ADOBE_ANALYTICS_CLIENT_SECRET")
+    @exchange_jwt_url = ENV.fetch("ADOBE_ANALYTICS_EXCHANGE_JWT_URL")
+  end
+
+  def perform
+    return if analytics.actual?
```

Comment:
It'd be nice if all of the instance vars in this class would use `@`, e.g. `@analytics`

Reply:
I would appreciate explanation why it would be better. But ok, I'm changing this.

Reply:
It's much easier to see if the code is referring to an instance variable or a method call that way.